handler/src/main/java/io/netty/handler/ssl/OpenSslContext.java

```diff
@@ -501,8 +524,52 @@ public OpenSslEngine remove(long ssl) {
         public void add(OpenSslEngine engine) {
             engines.put(engine.sslPointer(), engine);
         }
+
+        @Override
+        public OpenSslEngine get(long ssl) {
+            return engines.get(ssl);
+        }
     }
 
+    static void setKeyMaterial(long ctx, X509Certificate[] keyCertChain, PrivateKey key, String keyPassword)
+            throws SSLException {
+         /* Load the certificate file and private key. */
+        long keyBio = 0;
+        long keyCertChainBio = 0;
+
+        try {
+            keyCertChainBio = toBIO(keyCertChain);
+            keyBio = toBIO(key);
+
+            if (!SSLContext.setCertificateBio(
+                    ctx, keyCertChainBio, keyBio, keyPassword, SSL.SSL_AIDX_RSA)) {
+                long error = SSL.getLastErrorNumber();
+                if (OpenSsl.isError(error)) {
+                    throw new SSLException("failed to set certificate and key: "
+                            + SSL.getErrorString(error));
+                }
+            }
+            // We may have more then one cert in the chain so add all of them now.
+            if (!SSLContext.setCertificateChainBio(ctx, keyCertChainBio, false)) {
+                long error = SSL.getLastErrorNumber();
```

Comment:
Its only void for SSL but not for SSLContext... I not want to change signature to not break stuff ... That said it will never return false anyway (it wil always throw) so let me remove the checks